io/plugins/eu.esdihumboldt.hale.io.xsd/src/eu/esdihumboldt/hale/io/xsd/reader/XmlSchemaReader.java

```diff
@@ -689,15 +692,26 @@ else if (item instanceof XmlSchemaNotation) {
 				XmlSchemaExternal imp = (XmlSchemaExternal) externalItems.getItem(i);
 				XmlSchema importedSchema = imp.getSchema();
 				String location = importedSchema.getSourceURI();
-				if (!(imports.contains(location))) { // only add schemas that
+				String targetNamespace = importedSchema.getTargetNamespace();
+				if (!imports.containsKey(location)) { // only add schemas that
```

Comment:
Is it an issue that they key is a String (see line 362) and the `location` it is now comparted to might be a URI?

Reply:
@JohannaOtt The variable `location` is also of type String (not URI), so there shouldn't be a problem, I think. Is that what you meant?

Reply:
Sorry, misread the line. Yes, that is what I meant.